### DISCUSSION TITLE
Shelly plus add on possible sensors (fingerprint/ Motion)?

### DISCUSSION THREAD

**Discussion by kroon040** [arendst/Tasmota#20607] [Q&A]
Hi,

I noticed in another discussion #20580 that the shelly plus addo supports temperature sensors.

Is it possible that gpio1 and gpio19 can be used for the fingerprint sensor (r503/as608) for rx and tx?

Also the motion sensor HC-SR501, can this als be used on the addon?

**Comment by arendst** [comment]
I very much doubt it.

The Add-On is not a pass-through device. Ie the onewire protocol, normally using one wire (what's in a name) needs two GPIO's with this device because it can only do output on GPIO0 and input on GPIO1. GPIO3 is connected to a sawtooth comparator providing some relation between the analog voltage at it's input. Configure a counter on GPIO3 and you may be able to use it. Finally GPIO19 can only be used as input for a button or a switch and possibly a counter too.

If someone manages to get serial comms functional pls let me know.

**Comment by kroon040** [comment]
Ok for the fingerprint I understand, and for the motion sensor?? I see in this link it should work with default firmware??

https://cdn.vincenzocaputo.com/getimage/i=8068-c=89178-h=905-q=80/shelly-plus-add-on-schemi-di-montaggio-sensori-compatibili.webp

**Reply by arendst** [reply]
Give it a try and let us know if it works.
